### DISCUSSION TITLE
Reading CSV from a string?

### DISCUSSION THREAD

**Discussion by lukerobi** [JoshClose/CsvHelper#2151] [Q&A]
I have a vendor that wants to send me CSV data through an API, so it will be a string.  How could I parse this into something iterable?  Everything I have seen so far requires me to write it to a file, then read it, which seems cumbersome, slow, and completely unnecessary..

**Comment by JoshClose** [comment]
You can use `StringReader`.

ex:
```cs
void Main()
{
	var s = new StringBuilder();
	s.Append("Id,Name\r\n");
	s.Append("1,one\r\n");
	s.Append("2,two\r\n");
	var config = new CsvConfiguration(CultureInfo.InvariantCulture)
	{
	};
	using (var reader = new StringReader(s.ToString()))
	using (var csv = new CsvReader(reader, config))
	{
		csv.Context.RegisterClassMap<FooMap>();
		csv.GetRecords<Foo>().ToList().Dump();
	}
}

public class Foo
{
	public int Id { get; set; }
	public string Name { get; set; }
}

public sealed class FooMap : ClassMap<Foo>
{
	public FooMap()
	{
		Map(m => m.Id);
		Map(m => m.Name);
	}
}
```

**Reply by lukerobi** [reply]
Thank you very much for the information - I've never used an approach like this, so time to go off and try something new!
